system/Helpers/url_helper.php

```diff
@@ -121,37 +132,59 @@ function site_url($relativePath = '', ?string $scheme = null, ?App $config = nul
      */
     function base_url($relativePath = '', ?string $scheme = null): string
     {
-        $config            = clone config('App');
-        $config->indexPage = '';
+        $uri = _get_uri($relativePath);
 
-        return site_url($relativePath, $scheme, $config);
+        return URI::createURIString(
+            $scheme ?? $uri->getScheme(),
+            $uri->getAuthority(),
+            $uri->getPath(),
+            $uri->getQuery(),
+            $uri->getFragment()
+        );
```

Comment:
For me, the previous code was more readable.

Reply:
I need to refactor the whole three function code because I cannot fix without adding an parameter.
Because the previous code passes the Config object to `site_url()`.
So the Config value is always used and `base_url()` cannot support multiple domain.

Reply:
Ok, thanks - now I see the problem.

Since we're cloning the App config here... can't we use something like this?
```php
$config            = clone config('App');
$config->baseURL   = Services::request()->getUri()->getBaseURL();
$config->indexPage = '';

return site_url($relativePath, $scheme, $config);
```

Reply:
Nice!

I refactored and we are back to almost the same code as before.